packages/block-library/src/cover/edit/inspector-controls.js

```diff
@@ -42,46 +39,29 @@ function CoverHeightInput( {
 	unit = 'px',
 	value = '',
 } ) {
-	const instanceId = useInstanceId( UnitControl );
-	const inputId = `block-cover-height-input-${ instanceId }`;
-	const isPx = unit === 'px';
-
-	const [ availableUnits ] = useSettings( 'spacing.units' );
-	const units = useCustomUnits( {
-		availableUnits: availableUnits || [ 'px', 'em', 'rem', 'vw', 'vh' ],
-		defaultValues: { px: 430, '%': 20, em: 20, rem: 20, vw: 20, vh: 50 },
-	} );
-
 	const handleOnChange = ( unprocessedValue ) => {
-		const inputValue =
-			unprocessedValue !== ''
-				? parseFloat( unprocessedValue )
-				: undefined;
+		// eslint-disable-next-line @wordpress/no-unused-vars-before-return
```

Comment:
I'd like to avoid disabling ESLint errors if possible. Can I use logic like the following to deal with this?

```javascript
const parsedQuantityAndUnit =
	parseQuantityAndUnitFromRawValue( unprocessedValue );
const currentValue = parsedQuantityAndUnit[ 0 ];

if ( isNaN( currentValue ) && currentValue !== undefined ) {
	return;
}

const currentUnit = parsedQuantityAndUnit[ 1 ];

onUnitChange( currentUnit );
onChange( currentValue );
```

This is similar to [this part](https://github.com/WordPress/gutenberg/blob/7586f2f0904f64ab64d63d4d1853981a530c9f34/packages/components/src/unit-control/test/index.tsx#L41-L49).